compiler/rustc_hir_analysis/src/errors.rs

```diff
@@ -1459,6 +1459,8 @@ pub struct StaticMutRef {
     #[primary_span]
     #[label]
     pub span: Span,
+    #[note(hir_analysis_note_mut)]
+    pub note_mut: (),
```

Comment:
What's up with these `()` fields? Elsewhere in the compiler, such notes are managed entirely with attributes and no fields.
@davidtwco what is the recommended way here?